File: find-all-lonely-numbers-in-the-array/find-all-lonely-numbers-in-the-array.py
class Solution:
    def findLonely(self, nums: List[int]) -> List[int]:
        num = Counter(nums)
        res = []
        for key , val in num.items():
            if val==1 and key+1 not in num and key-1 not in num:
                res.append(key)
        return res

File: find-all-lonely-numbers-in-the-array/find-all-lonely-numbers-in-the-array.cpp
class Solution {
public:
    vector<int> findLonely(vector<int>& nums) {
        map<int,int>mp ;
        vector<int>res  ; 
        for(int i : nums){
            mp[i]++;
        }
        for(int i = 0 ; i < nums.size() ; i++){
            if(mp[nums[i]]==1 && mp.find(nums[i]-1)==mp.end() && mp.find(nums[i]+1)==mp.end()){
                res.push_back(nums[i]);
            }
        }
        return res;
    }
};